File: problem_2/problem_2.clj
(def fib-sequence
  (lazy-cat [0 1] (map +
                       (rest fib-sequence)
                       fib-sequence)))

(println
  (reduce +
          (filter even?
                  (take-while
                    (partial > 4000000)
                    fib-sequence))))

File: problem_2/problem_2.rb
def generate_fib_seq
  fib = [0, 1]

  while fib.last < 4_000_000
    fib << fib.last(2).reduce(:+)
  end

  fib
end

fib_seq = generate_fib_seq
sum = fib_seq.select {|f| f.even?}.reduce(:+)

puts sum
